decorator.rb

```diff
@@ -0,0 +1,29 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError
+  end
+end
+
+class Decorator < Nameable
+  super class Nameable
+          def initialize(nameable)
+            @nameable = nameable
+          end
+        end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.upcase
+  end
+end
+
+class TrimDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.slice(0, 10) unless @nameable.correct_name.length <= 10
+  end
+end
```

Comment:
- [ ] **[OPTIONAL]** In keeping with Ruby conventions, classes should be in different files. I would recommend you re-organize your classes accordingly.